src/reports/biospecimen-request/index.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+import { NextFunction, Request, Response } from 'express';
+import format from 'date-fns/format';
+import * as fs from 'fs';
+import JSZip from 'jszip';
+import EsInstance from '../../ElasticSearchClientInstance';
+import { reportGenerationErrorHandler } from '../../errors';
+import config from './config';
+import { normalizeConfigs } from '../../utils/configUtils';
+import ExtendedReportConfigs from '../../utils/extendedReportConfigs';
+import { createSet } from '../../utils/userClient';
+import { getUTCDate } from '../../utils/dateUtils';
+import generateFiles from './generateBiospecimenRequestFiles';
+
+const biospecimenRequest = async (req: Request, res: Response, _next: NextFunction): Promise<void> => {
+    console.time('biospecimenRequest');
+
+    const { sqon, projectId, biospecimenRequestName } = req.body;
+    const userId = req['kauth']?.grant?.access_token?.content?.sub;
+    const accessToken = req.headers.authorization;
+
+    const esClient = EsInstance.getInstance();
+
+    try {
+        await createSet(userId, accessToken, projectId, sqon, biospecimenRequestName);
+
+        // decorate the configs with default values, values from arranger's project, etc...
+        const normalizedConfigs: ExtendedReportConfigs = await normalizeConfigs(esClient, projectId, config);
+
+        const nowUTC = getUTCDate();
+        const filenameZip = `include_biospecimenRequest_${format(nowUTC, "yyyyMMdd'T'HHmmss'Z'")}.zip`;
```

Comment:
I think you should encapsulate this bit of code in a funct
Something like:
`const makePath = (ext: string) => `include_biospecimenRequest_README_${format(nowUTC, "yyyyMMdd'T'HHmmss'Z'")}.${ext}``